test/unittests/test_dd_synthesis.cpp

```diff
@@ -0,0 +1,551 @@
+
+    TruthTable::Cube c00{};
+    TruthTable::Cube c01{};
+    TruthTable::Cube c10{};
+    TruthTable::Cube c11{};
+
+    TruthTable::Cube c000{};
+    TruthTable::Cube c001{};
+    TruthTable::Cube c010{};
+    TruthTable::Cube c011{};
+    TruthTable::Cube c100{};
+    TruthTable::Cube c101{};
+    TruthTable::Cube c110{};
+    TruthTable::Cube c111{};
+
+    TruthTable::Cube c0000{};
+    TruthTable::Cube c0001{};
+    TruthTable::Cube c0010{};
+    TruthTable::Cube c0011{};
+    TruthTable::Cube c0100{};
+    TruthTable::Cube c0101{};
+    TruthTable::Cube c0110{};
+    TruthTable::Cube c0111{};
+    TruthTable::Cube c1000{};
+    TruthTable::Cube c1001{};
+    TruthTable::Cube c1010{};
+    TruthTable::Cube c1011{};
+    TruthTable::Cube c1100{};
+    TruthTable::Cube c1101{};
+    TruthTable::Cube c1110{};
+    TruthTable::Cube c1111{};
+
+    TruthTable::Cube c000000{};
+    TruthTable::Cube c000001{};
+    TruthTable::Cube c000010{};
+    TruthTable::Cube c000011{};
+    TruthTable::Cube c000100{};
+    TruthTable::Cube c000101{};
+    TruthTable::Cube c000110{};
+    TruthTable::Cube c000111{};
+    TruthTable::Cube c001000{};
+    TruthTable::Cube c001001{};
+    TruthTable::Cube c001010{};
+    TruthTable::Cube c001011{};
+    TruthTable::Cube c001100{};
+    TruthTable::Cube c001101{};
+    TruthTable::Cube c001110{};
+    TruthTable::Cube c001111{};
+
+    TruthTable::Cube c010000{};
+    TruthTable::Cube c010001{};
+    TruthTable::Cube c010010{};
+    TruthTable::Cube c010011{};
+    TruthTable::Cube c010100{};
+    TruthTable::Cube c010101{};
+    TruthTable::Cube c010110{};
+    TruthTable::Cube c010111{};
+    TruthTable::Cube c011000{};
+    TruthTable::Cube c011001{};
+    TruthTable::Cube c011010{};
+    TruthTable::Cube c011011{};
+    TruthTable::Cube c011100{};
+    TruthTable::Cube c011101{};
+    TruthTable::Cube c011110{};
+    TruthTable::Cube c011111{};
+
+    TruthTable::Cube c100000{};
+    TruthTable::Cube c100001{};
+    TruthTable::Cube c100010{};
+    TruthTable::Cube c100011{};
+    TruthTable::Cube c100100{};
+    TruthTable::Cube c100101{};
+    TruthTable::Cube c100110{};
+    TruthTable::Cube c100111{};
+    TruthTable::Cube c101000{};
+    TruthTable::Cube c101001{};
+    TruthTable::Cube c101010{};
+    TruthTable::Cube c101011{};
+    TruthTable::Cube c101100{};
+    TruthTable::Cube c101101{};
+    TruthTable::Cube c101110{};
+    TruthTable::Cube c101111{};
+
+    TruthTable::Cube c110000{};
+    TruthTable::Cube c110001{};
+    TruthTable::Cube c110010{};
+    TruthTable::Cube c110011{};
+    TruthTable::Cube c110100{};
+    TruthTable::Cube c110101{};
+    TruthTable::Cube c110110{};
+    TruthTable::Cube c110111{};
+    TruthTable::Cube c111000{};
+    TruthTable::Cube c111001{};
+    TruthTable::Cube c111010{};
+    TruthTable::Cube c111011{};
+    TruthTable::Cube c111100{};
+    TruthTable::Cube c111101{};
+    TruthTable::Cube c111110{};
+    TruthTable::Cube c111111{};
```

Comment:
This is truly horrible. Now that you implemented a PLA parser, it is far easier to specify the truth tables in PLA format as strings in the respective tests. Something along the lines of (untested):
```
auto pla = "... pla definition";
auto ss = std::stringstream(pla);
TruthTable tt{};
read_pla(tt, ss);
```

Reply:
I understand that. But since the PLA parser is implemented now, this can be done properly.